src/transformers/models/tapas/modeling_tapas.py

```diff
@@ -0,0 +1,2317 @@
+            elif scope_names[0] == "output_weights_agg":
+                pointer = getattr(pointer, "aggregation_classifier")
+                pointer = getattr(pointer, "weight")
+            # classification head
+            elif scope_names[0] == "output_bias_cls":
+                pointer = getattr(pointer, "classifier")
+                pointer = getattr(pointer, "bias")
+            elif scope_names[0] == "output_weights_cls":
+                pointer = getattr(pointer, "classifier")
+                pointer = getattr(pointer, "weight")
+            else:
+                try:
+                    pointer = getattr(pointer, scope_names[0])
+                except AttributeError:
+                    logger.info("Skipping {}".format("/".join(name)))
+                    continue
+            if len(scope_names) >= 2:
+                num = int(scope_names[1])
+                pointer = pointer[num]
+        if m_name[-11:] == "_embeddings":
+            pointer = getattr(pointer, "weight")
+        elif m_name[-13:] in [
+            "_embeddings_0",
+            "_embeddings_1",
+            "_embeddings_2",
+            "_embeddings_3",
+            "_embeddings_4",
+            "_embeddings_5",
+            "_embeddings_6",
+        ]:
+            pointer = getattr(pointer, "weight")
+        elif m_name == "kernel":
+            array = np.transpose(array)
+        try:
+            assert (
+                pointer.shape == array.shape
+            ), f"Pointer shape {pointer.shape} and array shape {array.shape} mismatched"
+        except AssertionError as e:
+            e.args += (pointer.shape, array.shape)
+            raise
+        logger.info("Initialize PyTorch weight {}".format(name))
+        # added a check to see whether the array is a scalar (because bias terms in Tapas checkpoints can be scalar => should first be converted to numpy arrays)
+        if np.isscalar(array):
+            array = np.array(array)
+        pointer.data = torch.from_numpy(array)
+    return model
+
+
+class TapasEmbeddings(nn.Module):
+    """
+    Construct the embeddings from word, position and token_type embeddings. Same as BertEmbeddings but with a number of
+    additional token type embeddings to encode tabular structure.
+    """
+
+    def __init__(self, config):
+        super().__init__()
+        # we do not include config.disabled_features and config.disable_position_embeddings from the original implementation
+        # word embeddings
+        self.word_embeddings = nn.Embedding(config.vocab_size, config.hidden_size, padding_idx=config.pad_token_id)
+        # position embeddings
+        self.position_embeddings = nn.Embedding(config.max_position_embeddings, config.hidden_size)
+        # token type embeddings
+        token_type_embedding_name = "token_type_embeddings"
+
+        for i, type_vocab_sizes in enumerate(config.type_vocab_sizes):
+            name = "%s_%d" % (token_type_embedding_name, i)
```

Comment:
(nit) 
```suggestion
            name = f"token_type_embeddings_{i}" 
```
and delete line 289

Reply:
Hmm this is really weird, I actually already applied this suggested change as can be seen [here](https://github.com/huggingface/transformers/pull/8113/commits/036326498fe5a366162b4927640fe2dbcdbf6ae0) (line 326).

@LysandreJik did you perhaps revert this?

Reply:
It is possible you didn't use that commit when doing the migration to the v4 version of your branch. Looking at the [git blame](https://github.com/NielsRogge/transformers/blame/tapas_v4/src/transformers/models/tapas/modeling_tapas.py#L292), the single commit that you made was "First commit: adding all files from tapas_v3", which did not contain this change, whereas this change can be seen in the [final version of your v3 branch](https://github.com/NielsRogge/transformers/blame/tapas_v3/src/transformers/modeling_tapas.py#L327)

Reply:
When copying all the files of the `tapas_v3` branch to `tapas_v4`, I did not update `tapas_v3` anymore. I see that my last commit on `tapas_v3` was 8 days ago, and the "First commit: adding all files from tapas_v3" was 7 days ago. So not sure why this change was not included 🤔. I hope it's the only change that is not included

Reply:
Oh I get now why! It's because I've only done it at one place, but @patrickvonplaten suggests this change at two places. Nevermind :)